src/main/java/net/coderbot/iris/shaderpack/LegacyIdMap.java

```diff
@@ -56,31 +57,42 @@ public static void addLegacyValues(Object2IntMap<BlockState> blockIdMap) {
 				Blocks.RED_STAINED_GLASS_PANE, Blocks.BLACK_STAINED_GLASS_PANE);
 
 		// Short grass / bush
-		add(blockIdMap, 31, Blocks.GRASS, Blocks.SEAGRASS, Blocks.SWEET_BERRY_BUSH);
+		addBlock(blockIdMap, 31, Blocks.GRASS, Blocks.SEAGRASS, Blocks.SWEET_BERRY_BUSH);
 
 		// Crops (59 = wheat), but we include carrots and potatoes too.
-		add(blockIdMap, 59, Blocks.WHEAT, Blocks.CARROTS, Blocks.POTATOES);
+		addBlock(blockIdMap, 59, Blocks.WHEAT, Blocks.CARROTS, Blocks.POTATOES);
 
 		// Small flowers
-		add(blockIdMap, 37, Blocks.DANDELION, Blocks.POPPY, Blocks.BLUE_ORCHID, Blocks.ALLIUM, Blocks.AZURE_BLUET,
+		addBlock(blockIdMap, 37, Blocks.DANDELION, Blocks.POPPY, Blocks.BLUE_ORCHID, Blocks.ALLIUM, Blocks.AZURE_BLUET,
 				Blocks.RED_TULIP, Blocks.PINK_TULIP, Blocks.WHITE_TULIP, Blocks.ORANGE_TULIP, Blocks.OXEYE_DAISY,
 				Blocks.CORNFLOWER, Blocks.LILY_OF_THE_VALLEY, Blocks.WITHER_ROSE);
 
 		// Big tall grass / flowers
 		// Also include seagrass here
-		add(blockIdMap, 175, Blocks.SUNFLOWER, Blocks.LILAC, Blocks.TALL_GRASS, Blocks.LARGE_FERN, Blocks.ROSE_BUSH,
+		addBlock(blockIdMap, 175, Blocks.SUNFLOWER, Blocks.LILAC, Blocks.TALL_GRASS, Blocks.LARGE_FERN, Blocks.ROSE_BUSH,
 				Blocks.PEONY, Blocks.TALL_SEAGRASS);
 
 		// Fire
-		add(blockIdMap, 51, Blocks.FIRE);
+		addBlock(blockIdMap, 51, Blocks.FIRE);
 
 		// Lily pad
-		add(blockIdMap, 111, Blocks.LILY_PAD);
+		addBlock(blockIdMap, 111, Blocks.LILY_PAD);
 	}
 
-	private static void add(Object2IntMap<BlockState> blockIdMap, int id, Block... leaves) {
-		for (Block leaf : leaves) {
-			leaf.getStateManager().getStates().forEach(state -> blockIdMap.put(state, id));
+	public static void addVanillaDimensions(Object2IntMap<Identifier> dimensionIdMap) {
+		addDimension(dimensionIdMap, 0, new Identifier("overworld"));
+		addDimension(dimensionIdMap, -1, new Identifier("the_nether"));
+		addDimension(dimensionIdMap, 1, new Identifier("the_end"));
+	}
+
+	private static void addBlock(Object2IntMap<BlockState> blockIdMap, int id, Block... blocks) {
+		for (Block block : blocks) {
+			block.getStateManager().getStates().forEach(state -> blockIdMap.put(state, id));
 		}
 	}
+
+	private static void addDimension(Object2IntMap<Identifier> dimensionIdMap, int id, Identifier dimension) {
+		dimensionIdMap.defaultReturnValue(0);
```

Comment:
This isn't valid - unknown dimensions should be mapped to the root directory, not to world0.